pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp

```diff
@@ -217,7 +218,10 @@ sample(const at::Tensor& rowptr,
   AT_DISPATCH_INTEGRAL_TYPES(seed.scalar_type(), "sample_kernel", [&] {
     typedef std::pair<scalar_t, scalar_t> pair_scalar_t;
     typedef std::conditional_t<!disjoint, scalar_t, pair_scalar_t> node_t;
-    typedef NeighborSampler<node_t, scalar_t, replace, directed, return_edge_id>
+    // TODO(zeyuan): Do not force int64_t for time type.
+    typedef int64_t temporal_t;
```

Comment:
Would it be better to check type equality (`static_assert` + `std::is_same`) and error out at compile time instead of implicitly promoting `temporal_t` to `int64_t`?

Reply:
The callers will only pass (dict of) tensors as args, so it might not be done in a static way. If we want to error out before pointer casting, I could add tensor type checking in the first place. wdyt?

Reply:
I was instead thinking something like a static check of `temporal_t` within the template. Happy to chat further, not super important.